relay-event-normalization/src/lib.rs

```diff
@@ -207,3 +214,153 @@ impl<'a> Processor for StoreProcessor<'a> {
+    pub is_renormalize: bool,
+
+    /// When `true`, infers the device class from CPU and model.
+    pub device_class_synthesis_config: bool,
+
+    /// When `true`, extracts tags from event and spans and materializes them into `span.data`.
+    pub enrich_spans: bool,
+
+    /// When `true`, computes and materializes attributes in spans based on the given configuration.
+    pub light_normalize_spans: bool,
+
+    /// The maximum allowed size of tag values in bytes. Longer values will be cropped.
+    pub max_tag_value_length: usize, // TODO: move span related fields into separate config.
+
+    /// Configuration for replacing identifiers in the span description with placeholders.
+    ///
+    /// This is similar to `transaction_name_config`, but applies to span descriptions.
+    pub span_description_rules: Option<&'a Vec<SpanDescriptionRule>>,
+
+    /// Configuration for generating performance score measurements for web vitals
+    pub performance_score: Option<&'a PerformanceScoreConfig>,
+
+    /// An initialized GeoIP lookup.
+    pub geoip_lookup: Option<&'a GeoIpLookup>,
+
+    /// When `Some(true)`, individual parts of the event payload is trimmed to a maximum size.
+    ///
+    /// See the event schema for size declarations.
+    pub enable_trimming: bool,
+}
+
+impl<'a> Default for NormalizationConfig<'a> {
+    fn default() -> Self {
+        Self {
+            client_ip: Default::default(),
+            user_agent: Default::default(),
+            received_at: Default::default(),
+            max_secs_in_past: Default::default(),
+            max_secs_in_future: Default::default(),
+            transaction_range: Default::default(),
+            max_name_and_unit_len: Default::default(),
+            breakdowns_config: Default::default(),
+            normalize_user_agent: Default::default(),
+            transaction_name_config: Default::default(),
+            is_renormalize: Default::default(),
+            device_class_synthesis_config: Default::default(),
+            enrich_spans: Default::default(),
+            light_normalize_spans: Default::default(),
+            max_tag_value_length: usize::MAX,
+            span_description_rules: Default::default(),
+            performance_score: Default::default(),
+            geoip_lookup: Default::default(),
+            enable_trimming: false,
+            measurements: None,
+        }
+    }
+}
+
+/// Normalizes an event, rejecting it if necessary.
+///
+/// Normalization consists of applying a series of transformations on the event
+/// payload based on the given configuration.
+///
+/// The returned [`ProcessingResult`] indicates whether the passed event should
+/// be ingested or dropped.
+pub fn normalize_event(
+    event: &mut Annotated<Event>,
+    config: &NormalizationConfig,
+) -> ProcessingResult {
+    let Annotated(ref mut event, ref mut meta) = event;
+    let Some(ref mut event) = event else {
+        return Ok(());
+    };
```

Comment:
Will this also work here?

```suggestion
    let Annotated(Some(ref mut event), ref mut meta) = event else {
        return Ok(());
    };
```